### ISSUE TITLE
Server Main Menu accessibility

### ISSUE THREAD

**Issue by gtritchie** [rstudio/rstudio#4855]
# Overview
RStudio Server's menubar is hosted inside the HTML page (versus Desktop, whereas the main menu is "native" using Qt).

This menubar must be accessible via keyboard, and to screen readers. Guidance on menubars is at https://www.w3.org/TR/wai-aria-practices/#menu

The main menu is RStudio Server 1.2 and earlier is not accessible, with the following problems:

# Problems
All screen-reader tests performed with VoiceOver/Chrome/macOS 10.14.5, with QuickNav off (tap Left+Arrow and Right-Arrow)

- no keyboard shortcuts to get focus onto the menu bar and/or the individual menu items (File, etc.); for a keyboard-only user or a screenreader user this is a major impediment to learning and using RStudio
- tabbing into the menubar from outside the application reports "File Menu popup button 2 items". The menubar contains 11 menu items, not 2; it is actually treating a hidden text field as the first item, and the "File" menuitem as the second
- entering the menubar using VO+Left reads "row 1 menu bar 2 items, column 1 menubar 2 items; you are currently on a menubar inside of a cell."; all this cell/row business is due to incorrectly annotated table-based layout
- from File menu, use VO+Left arrow (or VO+Right arrow when menubar is selected), and focus goes onto a hidden text field used as part of the GWT main menu implementation ("edit text, blank, 1 of 2")
- same problem with hidden text field for menubars used for e.g. File menu, Edit menu, etc.
- down-arrow (or VO+Space then down-arrow) on closed File menu reads:  "New File, menu popup button, table 2 colums, 37 rows, you are currently on a table"); should be no mention of tables, and say that you are on a menubar, with correct number of items (37 rows includes separators, and hidden menu items)
- down-arrow to enter a menu, such as "Help", and use down-arrow to move down the menu; occasionally the highlight will disappear, and continuing to press down-arrow will cause it to reappear; this is due to focus going to hidden menu items
- Screen reader navigation within vertical menubars (e.g. File menu) not working, should be able to move with VO+Left and VO+Right
- Moving through menus with up/down arrow should skip disabled items, moving with Screen reader VO+Left/VO+Right should include disabled items and give verbal indication that they are disabled
- Main menu items should indicate if they are expanded or not via aria-expanded

**Comment by jmcphers** [comment]
Another accessibility-related change here (maybe tracked separately?) is that there needs to be a mechanism to skip past the menu in a screen reader ("skip navigation"). 

https://webaim.org/techniques/skipnav/

**Comment by gtritchie** [comment]
Yes, tracked separately. But not in Github at this time just in my very long detailed spec that I haven’t shared yet. I implemented such a thing already in the signin page. For the IDE itself I may leverage the skip nav hidden menu concept to also contain a shortcut to bring up the accessibility dialog.
